packages/core/src/application_ref.ts

```diff
@@ -682,20 +682,51 @@ export class ApplicationRef {
   }
 
   /**
-   * Bootstrap a new component at the root level of the application.
+   * Bootstrap a component onto the element identified by its selector or, optionally, to a
+   specified
+   * element
    *
    * @usageNotes
    * ### Bootstrap process
    *
-   * When bootstrapping a new root component into an application, Angular mounts the
-   * specified application component onto DOM elements identified by the componentType's
-   * selector and kicks off automatic change detection to finish initializing the component.
+   * When bootstrapping an application with a component, Angular mounts the specified application
+   * component onto a DOM element, and kicks off automatic change detection to finish initializing
+   * the component.
+   *
+   * If not provided in the `rootSelectorOrNode` argument, the selector of the provided
+   * `componentOrFactory` will be used to match the DOM elements.
+   *
+   * Please note that if multiple elements match the selector, the first matched element will be
+   * used as a host element.
+   *
+   * ### Example
+   *
+   * Generally, we define the component to bootstrap in the bootstrap array of `NgModule`,
+   * but it requires us to know the component while writing the application code.
+   *
+   * Imagine a situation where we have to wait for an API call to decide about the component to
+   * bootstrap. We can use the `ngDoBootstrap` hook of the `NgModule` and call this method to
+   * dynamically bootstrap a component.
+   *
+   * {@example core/ts/platform/platform.ts region='componentSelector'}
    *
    * Optionally, a component can be mounted onto a DOM element that does not match the
    * componentType's selector.
    *
-   * ### Example
-   * {@example core/ts/platform/platform.ts region='longform'}
+   * ```typescript
+   * ngDoBootstrap(appRef: ApplicationRef) {
+   *   // it can be a CSS selector
+   *   . . .
+   *   appRef.bootstrap(this.componentMap[componentName], '#root-element');
```

Comment:
I think we should avoid using `this.componentMap` here if possible, so it's easier to understand what the code does. 
```suggestion
   *   appRef.bootstrap(ComponentOne, '#root-element');
```
and in the `ngDoBootstrap` code below you can have an if-else block:
```
ngDoBootstrap(appRef: ApplicationRef) {
  this.fetchDataFromApi().then((componentName: string) => {
    if (componentName === 'one') {
      // #docregion elementSelector
      appRef.bootstrap(ComponentOne, '#root-element');
      // #enddocregion elementSelector
    } else {
      // #docregion elementInstance
      const element = document.querySelector('#root-element');
      appRef.bootstrap(ComponentTwo, element);
      // #enddocregion elementInstance
    }
  });
}
```

Reply:
@AndrewKushnir Thank you for the kind feedback.

For the `ngDoBootstrap` code, I think it will be better if we follow what @petebacondarwin mentioned above:

https://github.com/angular/angular/pull/42301#discussion_r638534546

Reply:
I agree that for the examples that are demonstrating the additional parameter, we can avoid the dynamic nature of the previous example:

```ts
export class AppModule2 {
  // #docregion elementSelector
  ngDoBootstrap(appRef: ApplicationRef) {
      appRef.bootstrap(AppComponent, '#root-element');
  }
  // #enddocregion elementSelector
}

export class AppModule3 {
  // #docregion elementInstance
  ngDoBootstrap(appRef: ApplicationRef) {
      const element = document.querySelector('#root-element');
      appRef.bootstrap(AppComponent, element);
    });
  }
  // #enddocregion elementInstance
}
```